### PR TITLE
Fix/searchbar position

### PR DESCRIPTION
**Pull request by inderharrysingh**
Fixes: #183 

This fixes this misaligment of the search bar without affecting the responsive ness

### How this works 
I have added conditional grid styling for the navbar which centrally aligned searchbox , 

### Demo
https://www.loom.com/share/500fb361b06744828005cd202e258015

### CODE REVIEW

**Comment by max-programming** [comment]
@inderharrysingh Looks good!
But on the homepage the navbar looks worse

image

**Comment by inderharrysingh** [comment]
@max-programming ohh! i see , let me fix it

**Comment by inderharrysingh** [comment]
Fixed it 
### Solution
Returning a  placeholder with the styles of navbar if  we are on main page 

image

---
image

File: components/Header.tsx [hunk 1]
```diff
@@ -11,24 +11,26 @@ import Search, { SearchProps } from './Search';
 const SearchBar = (props: SearchProps) => {
   const router = useRouter();
 
-  return (router.pathname === '/repos/[language]' && (
+  return (router.pathname === '/repos/[language]' ?(
       <Search {...props} />
     )
+    : 
+     <div className="hidden sm:inline-flex flex-1 max-w-md px-6 md:ml-5"></div>
```
Comment (max-programming) [inline components/Header.tsx:18]: Can you use something else here because of this extra whitespace?

image

**Review by max-programming — CHANGES_REQUESTED** [review]

**Comment by inderharrysingh** [comment]
Removed that extra spacing

File: components/Header.tsx (hunk 1 above)
Comment (inderharrysingh) [reply components/Header.tsx]: Removed extra white space

File: components/Header.tsx [hunk 2]
```diff
@@ -11,24 +11,26 @@ import Search, { SearchProps } from './Search';
 const SearchBar = (props: SearchProps) => {
   const router = useRouter();
 
-  return (router.pathname === '/repos/[language]' && (
```
Comment (janaSunrise) [inline components/Header.tsx:14]: Is this change necessary? Doesn't look to me.

**Review by janaSunrise — COMMENTED** [review]
Mostly looks like formatting changes to me.

File: components/Header.tsx [hunk 3]
```diff
@@ -11,24 +11,26 @@ import Search, { SearchProps } from './Search';
...
+  return (router.pathname === '/repos/[language]' ?(
       <Search {...props} />
     )
+    : 
+     <div className={props as string}></div>
```
Comment (max-programming) [inline components/Header.tsx:18]: Instead of directly putting an object `as string` it's better to maybe put nothing or take `className` as a prop

**Review by max-programming — CHANGES_REQUESTED** [review]

**Review by max-programming — APPROVED** [review]

**Comment by max-programming** [comment]
@all-contributors please add @inderharrysingh for design, code
